### ISSUE TITLE
'Error' message: ''errors' is missing in props validation (react/prop-types)'

### ISSUE THREAD

**Issue by jalalat** [yannickcr/eslint-plugin-react#1115]
I am getting a lint error at this line when i am trying to access this.props.errors 

`{this.props.errors.map(err => <div key={err}>{err}</div>)}`

```
import React, { Component, PropTypes } from 'react';

class AuthForm extends Component {
	constructor(props) {
		super(props);
		this.state = { email: '', password: '' };
	}
	onSubmit(event) {
		event.preventDefault();
		const { email, password } = this.state;
		this.props.onSubmit({ email, password });
	}

	render() {
		return (
			<div className="row">
				<form onSubmit={() => this.onSubmit()} className="col s6">
					<div className="input-field">
						<input
							value={this.state.email} placeholder="Email"
							onChange={e => this.setState({ email: e.target.value })}
						/>
					</div>
					<div className="errors">
						{this.props.errors.map(err => <div key={err}>{err}</div>)}
					</div>
					<button className="btn">Submit</button>
				</form>
			</div>
		);
	}
}

AuthForm.propTypes = {
	onSubmit: PropTypes.func.isRequired
};

export default AuthForm;

```

I tried to find how i can define errors array and tried various ways as well like:

`errors: PropTypes.arrayOf(Error)` 

i got a different lint message:

> message: 'propType "errors" is not required, but has no corresponding defaultProp declaration. (react/require-default-props)'

Please help; i am trying to find a solution for this for past 1-2 days now.

**Comment by benstepp** [comment]
You should define your propTypes for all props that you use.

```jsx
AuthForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.array,
};
```

---

The different lint message is trying to tell you that you have a non-required propType defined, but no defaultProp for it. In your current code if you do not pass an errors array, the component will fail to render. For example in this simple test case, you will get a cannot read property map of undefined because this.props.errors does not exist.

```jsx
test('does stuff', ()  => {
  const onSubmit = jest.fn();
  const wrapper = shallow(<AuthForm onSubmit={onSubmit} />);
  // ...
});
```

The solution is to either provide a defaultProp for this.props.errors, or make the errors mandatory and update your tests / code to comply to the new requirement.

```jsx
AuthForm.defaultProps = {
  errors: [],
};

// or

AuthForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  errors: PropTypes.array.isRequired,
};
```

**Comment by ljharb** [comment]
Also, `PropTypes.arrayOf` takes a PropType validation function, not a constructor.'

This appears answered.

**Comment by jalalat** [comment]
Thanks a lot for your response. It solved my problem.
